packages/@aws-cdk/aws-stepfunctions-tasks/lib/sqs/send-message.ts

```diff
@@ -50,6 +57,140 @@ export interface SqsSendMessageProps extends sfn.TaskStateBaseProps {
   readonly messageGroupId?: string;
 }
 
+/**
+ * Attributes associated with a message sent to an SQS queue.
+ */
+export class SqsMessageAttributes {
+  /**
+   * Message attributes provided directly from task input. The value must be an
+   * object where values match the `SQS.MessageAttributeValue` schema.
+   *
+   * @param path the JSON path expression where the attributes are located.
+   *
+   * @returns a new `SqsMessageAttributes` instance.
+   *
+   * @see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_MessageAttributeValue.html
+   */
+  public static fromJsonPathAt(path: string): SqsMessageAttributes {
+    return new SqsMessageAttributes(sfn.TaskInput.fromJsonPathAt(path));
+  }
+
+  /**
+   * Message attributes provided by an object literal.
+   *
+   * @param obj the message attributes to use.
+   *
+   * @returns a new `SqsMessageAttributes` instance.
+   */
+  public static fromObject(obj: { [key: string]: SqsMessageAttributeValue }): SqsMessageAttributes {
+    const rendered: { [key: string]: any; } = {};
+    for (const [key, value] of Object.entries(obj)) {
+      rendered[key] = value._taskInput.value;
+    }
+
+    return new SqsMessageAttributes(sfn.TaskInput.fromObject(rendered));
+  }
+
+  /** @internal */
+  public readonly _taskInput: sfn.TaskInput;
+
+  private constructor(value: sfn.TaskInput) {
+    this._taskInput = value;
+  }
+}
+
+/**
+ * Message attribute values.
+ */
+export class SqsMessageAttributeValue {
```

Comment:
I'm not a huge fan of these method names. What do you think about:
- fromJsonPathAt
- fromBase64Literal
- fromBase64JsonPathAt
- fromNumberLiteral
- fromNumberJsonPathAt
- fromStringLiteral
- fromStringJsonPathAt

Reply:
we could also add `fromLiteral(value: any)` that attempts to choose the SQS type from the value that's passed in

Reply:
`from<Type>Literal` is awkward IMO when the value you are providing is NOT a literal (i.e: it's a variable reference / parameter / ...). This is why I intentionally chose to NOT have the `Literal` part here.

And I find `from<Type>JsonPathAt` to be awkward, too, because it seems to say the `<Type>` value is the `JsonPath`, which is not the case.

Reply:
Okay, how about:
- fromJsonPath()
- <Type>()
- <Type>FromJsonPath()

I just want a little bit more consistency and a bit less wordiness